uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
+        };
+      }
+    }
+
+    /**
+     * Returns whether a date is of a specific type
+     * half_day_am or half_day_pm
+     *
+     * @param  {string} name
+     * @param  {object} leaveRequest
+     * @param  {string} date
+     *
+     * @return {boolean}
+     */
+    function isDayType(name, leaveRequest, date) {
+      var dayType = dayTypes[name];
+
+      if (moment(date).isSame(leaveRequest.from_date)) {
+        return dayType.value == leaveRequest.from_date_type;
+      }
+
+      if (moment(date).isSame(leaveRequest.to_date)) {
+        return dayType.value == leaveRequest.to_date_type;
+      }
+    }
+
+    /**
+     * Returns whether a leaveRequest is pending approval
+     *
+     * @param  {object} leaveRequest
+     * @return {boolean}
+     */
+    function isPendingApproval(leaveRequest) {
+      var status = leaveRequestStatuses[leaveRequest.status_id];
+
+      return status.name === 'waiting_approval';
+    }
+
+    /**
+     * Loads the absence periods
+     *
+     * @return {Promise}
+     */
+    function loadAbsencePeriods() {
+      return AbsencePeriod.all()
+        .then(function (absencePeriods) {
+          vm.absencePeriods = absencePeriods;
+          vm.selectedPeriod = _.find(vm.absencePeriods, function (period) {
+            return !!period.current;
+          });
+        });
+    }
+
+    /**
+     * Loads the absence types
+     *
+     * @return {Promise}
+     */
+    function loadAbsenceTypes() {
+      return AbsenceType.all()
+        .then(function (absenceTypes) {
+          vm.absenceTypes = absenceTypes;
+        });
+    }
+
+    /**
+     * Loads the calendar data
+     *
+     * @return {Promise}
+     */
+    function loadCalendar() {
+      return Calendar.get(vm.contactId, vm.selectedPeriod.id)
+        .then(function (calendar) {
+          vm.calendar = setCalendarProps(calendar);
+        });
+    }
+
+    /**
+     * Loads the leave request day types
+     *
+     * @return {Promise}
+     */
+    function loadDayTypes() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
+        .then(function (dayTypesData) {
+          var iterator,
+            length = dayTypesData.length,
+            typesObj = {};
+
+          // convert to an object with name as key
+          for (iterator = 0; iterator < length; iterator++) {
+            typesObj[dayTypesData[iterator].name] = dayTypesData[iterator];
+          }
+          dayTypes = typesObj;
+        });
+    }
+
+    /**
+     * Loads all the leave requests
+     *
+     * @return {Promise}
+     */
+    function loadLeaveRequest() {
+      return LeaveRequest.all({
+        contact_id: vm.contactId,
+        from_date: {
+          from: vm.selectedPeriod.start_date
+        },
+        to_date: {
+          to: vm.selectedPeriod.end_date
+        }
+      }).then(function (leaveRequestsData) {
+          leaveRequests = leaveRequestsData.list;
+          return loadCalendar();
```

Comment:
loadcalender is side affect of this function, maybe it can be extracted or the function documentation/name change accordingly like `loadLeaveRequestsAndCalender`